documentation/internal/Hardening.md

```diff
@@ -422,3 +425,55 @@ Basically, `CronJob` runs the approval command above for every CSR according to
 ### Auto Approval Service
 
 It is possible to install the kubelet-csr-approver service. For more information, refer to [[kubelet-csr-approver](https://github.com/postfinance/kubelet-csr-approver)](https://github.com/postfinance/kubelet-csr-approver). This service approves CSR automatically when a CSR is created according to several settings. It is better to restrict nodes' IP addresses (`providerIpPrefixes` option) and FQDN templates (providerRegex). For more information, refer to the official documentation.
+
+## Disabling auto-mounting of tokens for service accounts
+
+To disable the auto-mounting of service account tokens, we need to create secrets associated with perticular service account and mount that secret as a volume to the pod's specification wherver necessary.
+
+To acchieve it, we have to follow below proedure -
+
+### Disable auto-mounting
+To disbale the auto-mounting of token add `automountServiceAccountToken: false` falg to the service account properties as shwon below -
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: ingress-nginx
+  namespace: ingress-nginx
+automountServiceAccountToken: false
+...
+```
+
+### Create secret
+Create a new kuebernetes secret of type `kubernetes.io/service-account-token` as below -
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ingress-nginx-token
+  namespace: ingress-nginx
+  annotations:
+    kubernetes.io/service-account.name: ingress-nginx
+type: kubernetes.io/service-account-token
+```
+
+### Mount the token thorugh secrets
+Edit POD specification and mount the secret as volume to the pod as below -
+
+```yaml
+...
+volumeMounts:
+  - name: ingress-nginx-token
+    mountPath: /var/run/secrets/kubernetes.io/serviceaccount
+
+...
+
+volumes:
+  - name: service-account-token
```

Comment:
The `name` doesn't match with the one in `volumeMounts`

Reply:
corrected the name.